File: algorithm/力扣热题100/21_合并两个有序链表.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function (list1, list2) {
    const dummy = { next: null }
    let p = dummy;
    while (list1 && list2) {
        if (list1.val < list2.val) {
            p.next = list1;
            list1 = list1.next
        } else {
            p.next = list2;
            list2 = list2.next
        }
        p = p.next
    }
    p.next = list1 || list2;
    return dummy.next
};

File: algorithm/力扣热题100/21_合并两个有序链表.go
package main

type ListNode3 struct {
	Val  int
	Next *ListNode3
}

func mergeTwoLists(list1 *ListNode3, list2 *ListNode3) *ListNode3 {
	dummy := &ListNode3{0, nil}
	p := dummy
	for list1 != nil && list2 != nil {
		if list1.Val < list2.Val {
			p.Next = list1
			list1 = list1.Next
		} else {
			p.Next = list2
			list2 = list2.Next
		}
		p = p.Next
	}
	if list1 != nil {
		p.Next = list1
	} else {
		p.Next = list2
	}
	return dummy.Next
}
